go-controller/pkg/util/net_unit_test.go

```diff
@@ -278,3 +279,115 @@ func TestMatchIPFamily(t *testing.T) {
 		})
 	}
 }
+
+func TestJoinHostPortInt32(t *testing.T) {
+	tests := []struct {
+		desc    string
+		inpHost string
+		inpPort int32
+		outExp  string
+	}{
+		{
+			desc:    "empty string for host name and a negative port number",
+			inpPort: -25,
+			outExp:  ":-25",
+		},
+		{
+			desc:    "valid non-zero length string for host name",
+			inpHost: "hostname",
+			inpPort: 22,
+			outExp:  "hostname:22",
+		},
+		{
+			desc:    "valid IPv4 address as host name",
+			inpHost: "192.168.1.15",
+			inpPort: 22,
+			outExp:  "192.168.1.15:22",
+		},
+		{
+			desc:    "valid IPv6 address as host name",
+			inpHost: "fd01::1234",
+			inpPort: 22,
+			outExp:  "[fd01::1234]:22",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			res := JoinHostPortInt32(tc.inpHost, tc.inpPort)
+			t.Log(res)
+			assert.Equal(t, res, tc.outExp)
+		})
+	}
+}
+
+func TestIPAddrToHWAddr(t *testing.T) {
+	tests := []struct {
+		desc  string
+		inpIP net.IP
+	}{
+		// TODO: The function may need to be modified to ensure that the input net.IP passed is not empty.
+		// Passing empty net.IP causes panic
+		/*{
+			desc:	"test empty instance of net.IP",
+			inpIP: net.IP{},
+		},*/
+		{
+			desc:  "test IPv4 instance of net.IP",
+			inpIP: ovntest.MustParseIP("192.168.1.5"),
+		},
+		{
+			desc:  "test IPv6 instance of net.IP",
+			inpIP: ovntest.MustParseIP("fd01::1234"),
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			res := IPAddrToHWAddr(tc.inpIP)
+			t.Log(res)
+			assert.Equal(t, res[0], byte(0x0A))
+			assert.Equal(t, res[1], byte(0x58))
```

Comment:
@vjayaramrh instead of checking just the 1st two bytes, I think we should be asserting that the returned `res` against the expected MAC. For example: for IP `192.168.1.5` check the result will be []byte{0x0a, 0x58, 0c0, 0xa8, 0x1 0x5}